slick-hikaricp/src/main/scala/slick/jdbc/hikaricp/HikariCPJdbcDataSource.scala

```diff
@@ -58,17 +58,20 @@ object HikariCPJdbcDataSource extends JdbcDataSourceFactory {
 
     // Infrequently used
 
-    // `initializationFailFast` is deprecated and should be replaced by
-    // `initializationFailTimeout`. See HikariCP docs for more information:
-    // https://github.com/brettwooldridge/HikariCP#infrequently-used
-    // But we still respect the value if it configured.
-    c.getBooleanOpt("initializationFailFast").foreach(hconf.setInitializationFailFast)
-
     // The default value for `initializationFailFast` was false, which means the pool
     // will not fail to start if there is a problem when connecting to the database.
     // To keep this behavior, we need to set `initializationFailTimeout` to -1 as
     // documented by HikariCP.
-    hconf.setInitializationFailTimeout(c.getMillisecondsOr("initializationFailTimeout", -1))
+    hconf.setInitializationFailTimeout(c.getMillisecondsOr("initializationFailTimeout", default = {
+      // `initializationFailFast` is deprecated and should be replaced by
+      // `initializationFailTimeout`. See HikariCP docs for more information:
+      // https://github.com/brettwooldridge/HikariCP#infrequently-used
+      // But for backwards compatibility we check for it if initializationFailTimeout is missing
+      c.getBooleanOpt("initializationFailFast").map { failFast =>
```

Comment:
It could be a good idea to print a warning when `initializationFailFast` is present so that we can instruct users to move away from this configuration.